Subject: [PATCH] Add quirk for the graphics adapter in some JSxx
From: Tony Breeds <tony@bakeyournoodle.com>
References: 461002 - LTC50817

These devices are set to 640x480 by firmware, swtiuch them to
800x600@60.

Signed-off-by: Tony Breeds <tony@bakeyournoodle.com>
Signed-off-by: Olaf Hering <olh@suse.de>
---
 drivers/video/aty/radeon_monitor.c |   12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/drivers/video/aty/radeon_monitor.c
+++ b/drivers/video/aty/radeon_monitor.c
@@ -862,6 +862,18 @@ void __devinit radeon_check_modes(struct
 			has_default_mode = 1;
  	}
 
+#ifdef CONFIG_PPC_PSERIES
+	if (!has_default_mode &&
+	    of_device_is_compatible(rinfo->of_node, "pci1002,515e") &&
+	    (machine_is_compatible("IBM,8844")) ) {
+		printk("Falling back to 800x600 on JSxx hardware\n");
+		if (fb_find_mode(&info->var, info, "800x600@60",
+				 info->monspecs.modedb,
+				 info->monspecs.modedb_len, NULL, 8) != 0)
+			has_default_mode = 1;
+	}
+#endif
+
 	/*
 	 * Still no mode, let's pick up a default from the db
 	 */
